jfr-mappers/src/main/java/com/newrelic/jfr/tosummary/GCHeapSummarySummarizer.java

```diff
@@ -0,0 +1,102 @@
+/*
+ *
+ *  * Copyright 2020 New Relic Corporation. All rights reserved.
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package com.newrelic.jfr.tosummary;
+
+import static com.newrelic.jfr.tosummary.BaseDurationSummarizer.DEFAULT_CLOCK;
+
+import com.newrelic.telemetry.Attributes;
+import com.newrelic.telemetry.metrics.Summary;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+import jdk.jfr.consumer.RecordedEvent;
+
+/**
+ * This class aggregates the duration of GCHeapSummary JFR events. For GC purposes they come in
+ * pairs.
+ */
+public final class GCHeapSummarySummarizer implements EventToSummary {
+
+  public static final String EVENT_NAME = "jdk.GCHeapSummary";
+  private static final String BEFORE = "Before GC";
+  private static final String AFTER = "After GC";
+
+  private final Map<Long, RecordedEvent> awaitingPairs = new HashMap<>();
```

Comment:
are GC events always guaranteed to come in pairs?

Reply:
Yes - there will be a before and and after event